x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/bucket_nesting_editor.tsx

```diff
@@ -32,35 +35,75 @@ export function BucketNestingEditor({
   const columns = Object.entries(layer.columns);
   const aggColumns = columns
     .filter(([id, c]) => id !== columnId && c.isBucketed)
-    .map(([value, c]) => ({ value, text: c.label }));
+    .map(([value, c]) => ({
+      value,
+      text: c.label,
+      fieldName: hasField(c) ? c.sourceField : '',
+    }));
 
   if (!column || !column.isBucketed || !aggColumns.length) {
     return null;
   }
 
+  const fieldName = hasField(column) ? column.sourceField : '';
```

Comment:
This is the same logic as above. Not a big deal. But might be something we can tidy up later.